mso/resource_mso_tenant_policies_dhcp_relay_policy.go

```diff
@@ -0,0 +1,257 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"providers": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dhcp_server_address": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.IsIPAddress,
+						},
+						"application_epg": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"external_epg": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"dhcp_server_vrf_preference": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			for index, provider := range d.Get("providers").(*schema.Set).List() {
+				providerMap := provider.(map[string]interface{})
+				appEPG, appEPGExists := providerMap["application_epg"].(string)
+				extEPG, extEPGExists := providerMap["external_epg"].(string)
+
+				if appEPGExists && extEPGExists && appEPG != "" && extEPG != "" {
+					return fmt.Errorf("only one of 'application_epg' or 'external_epg' can be set for a provider at index position: %d", index)
+				}
+			}
+			return nil
+		},
+	}
+}
+
+func setDHCPRelayPolicyData(d *schema.ResourceData, response *container.Container, templateId string) error {
+	d.SetId(fmt.Sprintf("templateId/%s/DHCPRelayPolicy/%s", templateId, models.StripQuotes(response.S("name").String())))
+	d.Set("template_id", templateId)
+	d.Set("name", models.StripQuotes(response.S("name").String()))
+	d.Set("description", models.StripQuotes(response.S("description").String()))
+	d.Set("uuid", models.StripQuotes(response.S("uuid").String()))
+
+	providersList, err := response.S("providers").Children()
+	if err != nil {
+		return err
+	}
+	providers := make([]interface{}, 0)
+	for _, provider := range providersList {
+		providerMap := map[string]interface{}{
+			"dhcp_server_address":        models.StripQuotes(provider.S("ip").String()),
+			"dhcp_server_vrf_preference": provider.S("useServerVrf").Data().(bool),
+		}
+
+		applicationEPG := models.StripQuotes(provider.S("epgRef").String())
+		if applicationEPG != "{}" {
+			providerMap["application_epg"] = applicationEPG
+		}
+
+		externalEPG := models.StripQuotes(provider.S("externalEpgRef").String())
+		if externalEPG != "{}" {
+			providerMap["external_epg"] = externalEPG
+		}
+
+		providers = append(providers, providerMap)
+	}
+	d.Set("providers", providers)
+	return nil
+}
+
+func buildMSOTenantPoliciesDHCPRelayPolicyPayload(d *schema.ResourceData) map[string]interface{} {
+	payload := map[string]interface{}{
+		"name":        d.Get("name").(string),
+		"description": d.Get("description").(string),
+		"providers":   []interface{}{},
+	}
+
+	if uuid, ok := d.GetOk("uuid"); ok {
+		payload["uuid"] = uuid.(string)
+	}
```

Comment:
is uuid required to be set for update operation the reason that it is included in payload?